### PR TITLE
Remove Base Token Address from Chain Registration

### PR DESCRIPTION
**Pull request by Raid5594**
# What ❔

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.

### CODE REVIEW

File: l1-contracts/contracts/bridgehub/Bridgehub.sol [hunk 1]
```diff
@@ -41,14 +42,15 @@ contract Bridgehub is IBridgehub, ReentrancyGuard, Ownable2StepUpgradeable, Paus
 
     /// @notice StateTransitionManagers that are registered, and ZKchains that use these STMs can use this bridgehub as settlement layer.
     mapping(address _stateTransitionManager => bool) public stateTransitionManagerIsRegistered;
+
     /// @notice we store registered tokens (for arbitrary base token)
-    mapping(address _baseToken => bool) public tokenIsRegistered;
+    mapping(address _baseToken => bool) public __DEPRECATED_tokenIsRegistered;
```
Comment (kelemeno) [inline l1-contracts/contracts/bridgehub/Bridgehub.sol:47]: You said its not final, but just writing it down: assetIdIsRegistered needs to be added.

File: l1-contracts/contracts/state-transition/chain-deps/facets/Getters.sol [hunk 2]
```diff
@@ -66,7 +67,14 @@ contract GettersFacet is ZkSyncHyperchainBase, IGetters, ILegacyGetters {
 
     /// @inheritdoc IGetters
     function getBaseToken() external view returns (address) {
-        return s.baseToken;
+        IBridgehub bridgehub = IBridgehub(s.bridgehub);
+        bytes32 baseTokenAssetId = bridgehub.baseTokenAssetId(s.chainId);
```
Comment (kelemeno) [inline l1-contracts/contracts/state-transition/chain-deps/facets/Getters.sol:71]: actually you can get baseTokenAssetId = s.basetTokenAssetId

File: l1-contracts/deploy-scripts/DeployL1.s.sol [hunk 3]
```diff
@@ -666,7 +666,6 @@ contract DeployL1Script is Script {
     function registerSharedBridge() internal {
         Bridgehub bridgehub = Bridgehub(addresses.bridgehub.bridgehubProxy);
         vm.startBroadcast(msg.sender);
-        bridgehub.addToken(ADDRESS_ONE);
```
Comment (kelemeno) [inline l1-contracts/deploy-scripts/DeployL1.s.sol:669]: this needs to be replace with addTokenAssetId

File: l1-contracts/deploy-scripts/RegisterHyperchain.s.sol [hunk 4]
```diff
@@ -128,30 +128,11 @@ contract RegisterHyperchainScript is Script {
         console.log("Using base token address:", config.baseToken);
     }
 
-    function registerTokenOnBridgehub() internal {
```
Comment (kelemeno) [inline l1-contracts/deploy-scripts/RegisterHyperchain.s.sol:131]: Do not delete, instead rewrite for assetId

File: l1-contracts/scripts/register-hyperchain.ts [hunk 5]
```diff
@@ -97,9 +97,6 @@ async function main() {
       await checkTokenAddress(baseTokenAddress);
       console.log(`Using base token address: ${baseTokenAddress}`);
       console.log(deployer.addresses.Bridgehub.BridgehubProxy);
-      if (!(await deployer.bridgehubContract(deployWallet).tokenIsRegistered(baseTokenAddress))) {
```
Comment (kelemeno) [inline l1-contracts/scripts/register-hyperchain.ts:100]: Same: do not delete, rewrite

File: l1-contracts/contracts/bridgehub/Bridgehub.sol [hunk 6]
```diff
@@ -329,6 +333,17 @@ contract Bridgehub is IBridgehub, ReentrancyGuard, Ownable2StepUpgradeable, Paus
                              Getters
     //////////////////////////////////////////////////////////////*/
 
+    // Asset Handlers' token address function, which takes assetId as input and return the token address
```
Comment (StanislavBreadless) [inline l1-contracts/contracts/bridgehub/Bridgehub.sol:336]: please use nat spec for comments

File: l1-contracts/src.ts/deploy-test-process.ts [hunk 7]
```diff
@@ -329,7 +333,7 @@ export class EraDeployer extends Deployer {
         protocolVersion: CONTRACTS_GENESIS_PROTOCOL_VERSION,
         admin: this.ownerAddress,
         validatorTimelock: ADDRESS_ONE,
-        baseToken: ETH_ADDRESS_IN_CONTRACTS,
+        baseTokenAssetId: encodeNTVAssetId(31337, ethers.utils.hexZeroPad(ETH_ADDRESS_IN_CONTRACTS, 32)),
```
Comment (StanislavBreadless) [inline l1-contracts/src.ts/deploy-test-process.ts:336]: better to use a constant for 31337

**Pull request merged**